test/instances/morpho-blue/MorphoBlue_wstETH_USDC_Hyperdrive.t.sol

```diff
@@ -0,0 +1,818 @@
+        uint256 baseProceeds = closeLong(bob, maturityTime, longAmount);
+
+        // Bob should receive almost exactly his bond amount.
+        assertLe(
+            baseProceeds,
+            longAmount.mulDown(ONE - hyperdrive.getPoolConfig().fees.flat)
+        );
+        assertApproxEqAbs(
+            baseProceeds,
+            longAmount.mulDown(ONE - hyperdrive.getPoolConfig().fees.flat),
+            2
+        );
+
+        // Ensure that the withdrawal was processed as expected.
+        verifyWithdrawal(
+            bob,
+            baseProceeds,
+            true,
+            totalSupplyAssetsBefore,
+            totalSupplySharesBefore,
+            bobBalancesBefore,
+            hyperdriveBalancesBefore
+        );
+    }
+
+    /// Short ///
+
+    function test_open_short_nonpayable() external {
+        vm.startPrank(bob);
+
+        // Ensure that sending ETH to `openLong` fails.
+        vm.expectRevert(IHyperdrive.NotPayable.selector);
+        hyperdrive.openShort{ value: 2e18 }(
+            1e18,
+            1e18,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+
+        // Ensure that Bob receives a refund when he opens a short with "asBase"
+        // set to false and sends ether to the contract.
+        vm.expectRevert(IHyperdrive.NotPayable.selector);
+        hyperdrive.openShort{ value: 0.5e18 }(
+            1e18,
+            1e18,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: false,
+                extraData: new bytes(0)
+            })
+        );
+    }
+
+    function test_round_trip_short_instantaneous(
+        uint256 _shortAmount
+    ) external {
+        // Bob opens a short with base.
+        _shortAmount = _shortAmount.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            hyperdrive.calculateMaxShort()
+        );
+        IERC20(hyperdrive.baseToken()).approve(
+            address(hyperdrive),
+            _shortAmount
+        );
+        (uint256 maturityTime, uint256 basePaid) = openShort(bob, _shortAmount);
+
+        // Get some balance information before the withdrawal.
+        (
+            uint256 totalSupplyAssetsBefore,
+            uint256 totalSupplySharesBefore
+        ) = getSupply();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Bob closes his long with base as the target asset.
+        uint256 baseProceeds = closeShort(bob, maturityTime, _shortAmount);
+
+        // Bob should receive less base than he paid since no time as passed
+        // and the fees are non-zero.
+        assertLt(baseProceeds, basePaid);
```

Comment:
okay so on these instantaneous tests you are just making sure that the proceeds are less than the basePaid. i think this is a better way to test it anyways. we should probably add this type of assertion to the other morpho tests if we havent already

Reply:
Yep, I added these to the other tests. I still kept the `assertApproxEqAbs`, but I added a `assertLt`.